illstack/cyprof.pyx

```diff
@@ -33,32 +33,39 @@ def periodic_bcs(np.ndarray posp,np.ndarray posh):
     return posp
 
 def add_ghost_particles(posc,vals,maxrad):
-    posc_ghosts = np.empty((0),float)
-    vals_ghosts = np.empty((0),float)
+    #posc_ghosts = np.empty((0),float) #we aren't filling these with anything?
+    #vals_ghosts = np.empty((0),float)
+    posc_ghosts= posc
+    vals_ghosts=vals
```

Comment:
This looks good seeing as np.empty wasn't working, @marcelo-alvarez  this could also be where we're getting extra particles factor of 3 more rather than 2.

Reply:
if you are initializing posc_ghosts to posc, then you need to skip the i = j = k = 0 image in the loop below, i.e. 
if [ i==0 and j==0 and k==0 ]: continue